_posts/2018/2018-01-07-extract-conditionals-into-well-named-methods.md

```diff
@@ -0,0 +1,97 @@
+---
+title: 'Extract conditionals into well-named methods'
+description: 'An extraction to a private method often adds clarity'
+layout: article
+category: ruby
+image:
+  base: '2018/extract-conditionals-into-well-named-methods'
+  alt: 'Jigsaw pieces'
+  source: 'https://unsplash.com/photos/3y1zF4hIPCg'
+  credit: 'Hans-Peter Gauster'
+---
+
+One of the simplest (and most effective) ways to refactor is to [extract a method](https://refactoring.com/catalog/extractMethod.html). The naming of an extracted method is a great tool for communicating the thinking behind the code.
+
+## Instead of…
+
+...using slightly complex statements in your conditionals.
+
+```ruby
+class BrightonCoffeeShop
+  def initialize(name)
+    @name = name
+  end
+
+  def good?
+    if @name == 'Starbucks' || @name == 'Costa' ||
+       @name == 'Barry’s Caff'
+      false
+    elsif @name == 'Coffee@33' || @name == 'Small Batch'
+      true
+    else
+      false
+    end
+  end
+end
+```
+
+This implementation explicitly returns `false` if the `name` is unknown. The only way to guarantee a decent coffee shop experience is to rule out places you’ve never heard of. Right? :-)
+
+
+## Use…
+
+...the Extract Method refactoring to move the logic into a `private` method, giving that method a name that clearly explains the concept of the code.
+
+```ruby
+class BrightonCoffeeShop
+  def initialize(name)
+    @name = name
+  end
+
+  def good?
+    clean? && local? && excellent_coffee?
+  end
+
+  private
+
+  def clean?
+    !filthy?
+  end
+
+  def excellent_coffee?
+    ['Coffee@33', 'Small Batch'].include?(@name)
+  end
+
+  def filthy?
+    @name == 'Barry’s Caff'
+  end
+
+  def local?
+    !national_chain?
+  end
+
+  def national_chain?
+    ['Starbucks', 'Costa'].include?(@name)
+  end
+end
+```
+
+
+## But why?
+
+By extracting a method you, by definition, name it. Our `#good?` method clearly states the specific criteria for a good coffee shop. The new private methods act as clear documentation for the class.
```

Comment:
Maybe avoid using 'clearly' and 'clear' so closely. Might change 'clearly' to 'explicitly', so maybe: Our 'good?' method _now_ explicitly states the specific criteria... (emphasis for you).

Or keep 'clearly' and just remove 'clear' in the second sentence: The new private methods act as documentation for the class.